readme.md

```diff
@@ -246,19 +249,153 @@ string incomingDynatraceStringTag = ...; // retrieve from incoming call metadata
 incomingRemoteCallTracer.SetDynatraceStringTag(incomingDynatraceStringTag);
 
 incomingRemoteCallTracer.Start();
+incomingRemoteCallTracer.SetProtocolName("MyRemoteCallProtocol");
 try
 {
-  incomingRemoteCallTracer.SetProtocolName("MyRemoteCallProtocol");
-  ProcessRemoteCall();
+    ProcessRemoteCall();
 }
 catch (Exception e)
 {
-  incomingRemoteCallTracer.Error(e.Message);
-  // handle or rethrow
+    incomingRemoteCallTracer.Error(e.Message);
+    // handle or rethrow
+}
+finally
+{
+    incomingRemoteCallTracer.End();
+}
+```
+
+### Trace messaging
+
+You can use the SDK to trace messages sent or received via messaging & queuing systems. When tracing messages, we distinguish between:
+
+* sending a message
+* receiving a message
+* processing a received message
+
+To trace an outgoing message, you need to create an `IMessagingSystemInfo` and call `TraceOutgoingMessage` with that instance:
+
+```csharp
+string serverEndpoint = "messageserver.example.com:1234";
+string topic = "my-topic";
+IMessagingSystemInfo messagingSystemInfo = oneAgentSdk
+    .CreateMessagingSystemInfo("MyCustomMessagingSystem", topic, MessageDestinationType.TOPIC, ChannelType.TCP_IP, serverEndpoint);
+
+IOutgoingMessageTracer outgoingMessageTracer = oneAgentSdk.TraceOutgoingMessage(messagingSystemInfo);
+
+outgoingMessageTracer.Start();
+try
+{
+    Message message = new Message();
+    message.Headers[OneAgentSdkConstants.DYNATRACE_MESSAGE_PROPERTYNAME] = outgoingMessageTracer.GetDynatraceByteTag();
+
+    SendResult result = MyMessagingSystem.SendMessage(message);
+
+    outgoingMessageTracer.SetCorrelationId(result.CorrelationId);
```

Comment:
correlationId needs to be generated by application and must be set on message before SendMessage(). i know: need to clarify that in the Java sample ...